cg/cli/workflow/balsamic/base.py

```diff
@@ -0,0 +1,335 @@
+""" Add CLI support to create config and/or start BALSAMIC """
+import gzip
+import logging
+import re
+import subprocess
+import sys
+import shutil
+from pathlib import Path
+
+import click
+from cg.apps import hk
+from cg.apps.balsamic.fastq import FastqHandler
+from cg.cli.workflow.balsamic.store import store as store_cmd
+from cg.cli.workflow.balsamic.deliver import deliver as deliver_cmd
+from cg.cli.workflow.get_links import get_links
+from cg.exc import LimsDataError, BalsamicStartError
+from cg.meta.workflow.balsamic import AnalysisAPI
+from cg.apps import tb, scoutapi, lims
+from cg.store import Store
+
+LOG = logging.getLogger(__name__)
+PRIORITY_OPTION = click.option(
+    "-p", "--priority", type=click.Choice(["low", "normal", "high"])
+)
+EMAIL_OPTION = click.option("-e", "--email", help="email to send errors to")
+SUCCESS = 0
+FAIL = 1
+
+
+@click.group(invoke_without_command=True)
+@PRIORITY_OPTION
+@EMAIL_OPTION
+@click.option(
+    "-c", "--case-id", "case_id", help="case to prepare and start an analysis for"
+)
+@click.option("--target-bed", required=False, help="Optional")
+@click.pass_context
+def balsamic(context, case_id, priority, email, target_bed):
+    """Cancer workflow """
+    context.obj["db"] = Store(context.obj["database"])
+    context.obj["hk_api"] = hk.HousekeeperAPI(context.obj)
+    context.obj["analysis_api"] = AnalysisAPI
+    context.obj["fastq_handler"] = FastqHandler
+    context.obj["gzipper"] = gzip
+    context.obj["tb"] = tb.TrailblazerAPI(context.obj)
```

Comment:
does the trailblazer api need to be passed to context here?

Reply:
the Store, the Housekeeper and Trailblazer has the context as its 'config' parameter, can we avoid it?

Reply:
Sure, but I wondered why it is added to context.obj["tb"], It doesn't seem do be used downstream from this command. 

I see now that tb is needed downstream in the store command, but here a new api is instantiated instead of the one in the context.

https://github.com/Clinical-Genomics/cg/blob/7af0de152c51d12cfb5a63f29cf6dd945fa626e9/cg/cli/workflow/balsamic/store.py#L20

Reply:
Well spotted. removed now